artemis/modules/sql_injection_detector.py

```diff
@@ -0,0 +1,209 @@
+                        )
+            else:
+                for param_batch in more_itertools.batched(URL_PARAMS, 10):
+                    for error_payload in Config.Modules.SqlInjectionDetector.SQL_INJECTION_ERROR_PAYLOADS:
+                        url_with_payload = self.create_url_with_batch_payload(
+                            url=current_url, param_batch=param_batch, payload=error_payload
+                        )
+
+                        if not self.contains_error(http_requests.get(current_url)) and self.contains_error(
+                            http_requests.get(url_with_payload)
+                        ):
+                            message.append(
+                                f"{url_with_payload}: It appears that this url is vulnerable to SQL Injection"
+                            )
+
+                    for sleep_payload in Config.Modules.SqlInjectionDetector.SQL_INJECTION_SLEEP_PAYLOADS:
+                        flags = []
+                        url_with_sleep_payload = self.create_url_with_batch_payload(
+                            url=current_url, param_batch=param_batch, payload=sleep_payload
+                        )
+                        for _ in range(3):
+                            if self.are_requests_time_efficient(current_url) and not self.are_requests_time_efficient(
+                                url_with_sleep_payload
+                            ):
+                                flags.append(True)
+                            else:
+                                flags.append(False)
+
+                        if all(flags):
+                            message.append(
+                                f"{url_with_sleep_payload}: It appears that this url is vulnerable to SQL Time Base Injection"
+                            )
+
+            for error_payload in Config.Modules.SqlInjectionDetector.SQL_INJECTION_ERROR_PAYLOADS:
+                headers = self.create_headers(payload=error_payload)
+                if not self.contains_error(http_requests.get(current_url)) and self.contains_error(
+                    http_requests.get(current_url, headers=headers)
+                ):
+                    message.append(
+                        f"{current_url}: It appears that this url is vulnerable to SQL Injection through HTTP Headers"
+                    )
+
+            for sleep_payload in Config.Modules.SqlInjectionDetector.SQL_INJECTION_SLEEP_PAYLOADS:
+                headers = self.create_headers(sleep_payload)
+                if self.are_requests_time_efficient(current_url) and not self.are_requests_time_efficient(
+                    current_url, headers=headers
+                ):
+                    message.append(
+                        f"{current_url}: It appears that this url is vulnerable to SQL Time Base Injection through HTTP Headers"
+                    )
+
+            if http_requests.get(current_url).status_code == 500:
```

Comment:
let's treat the case as error (so contains_error should check for this), not with separate logic